### DISCUSSION TITLE
empty forecast of deepVAR even though the model successfully trained

### DISCUSSION THREAD

**Discussion by ShirleyMgit** [awslabs/gluonts#2692] [Q&A] [answered]
1. I am training deepVAR model, I succeed to train the model but it fails to predict. using make_evaluation_predictions() results in empty list (after doing list(forcesat_it)). can someone please explain why is it happening? 
2. In a differennt scenario I can get the forcasts list but the MultivariateEvaluator(quantiles=[0.1, 0.5, 0.9]) does. not work, I get: stopIteration, why?
3. I would be happy to know how can I get the predicted probability distribution parameters

**Comment by abdulfatir** [comment, marked as answer]
Can you post a snippet of how you're evaluating the model?

**Reply by ShirleyMgit** [reply]
forecast_it, ts_it = make_evaluation_predictions(test_tf, predictor,  num_samples=100)
    print("EVALUATION is complete SUCCESS")
    forecasts = list(forecast_it)
    tss = list(ts_it)
    print("after the list convertor: EVALUATION is complete SUCCESS")
    
    evaluator = MultivariateEvaluator(quantiles=[0.1, 0.5, 0.9])
    agg_metrics, item_metrics = evaluator(ts_it, forecast_it, num_series=len(test_tf))
    print("METRICS retrieved SUCCESS")

**Reply by abdulfatir** [reply]
I see a potential issue with your code which could be the reason for your problem. You're consuming iterators using `list()` and then passing the iterators again to the evaluator. These are single use iterators. Either pass the lists to the evaluator as
```python
agg_metrics, item_metrics = evaluator(tss, forecasts, num_series=len(test_tf))
```
or do not convert the iterators to list first.

**Reply by ShirleyMgit** [reply]
thanks, is it possible to include the  "feat_static_cat" in deepVAR, from some reason I can train the model but than it fails again in transforming the forecasts into list:
forecast_it, ts_it = make_evaluation_predictions(test_tf, predictor,  num_samples=100)
    print("EVALUATION is complete SUCCESS")
    forecasts = list(forecast_it)
    tss = list(ts_it)
